pkg/joinserver/grpc.go

```diff
@@ -0,0 +1,38 @@
+// Copyright © 2019 The Things Network Foundation, The Things Industries B.V.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package joinserver
+
+import (
+	"context"
+
+	pbtypes "github.com/gogo/protobuf/types"
+	"go.thethings.network/lorawan-stack/pkg/ttnpb"
+)
+
+type jsServer struct {
+	JS *JoinServer
+}
+
+// GetJoinEuiPrefixes returns the JoinEuiPrefixes associated with the join server.
+func (srv jsServer) GetJoinEuiPrefixes(ctx context.Context, _ *pbtypes.Empty) (*ttnpb.JoinEUIPrefixes, error) {
+	prefixes := make([]ttnpb.JoinEUIPrefix, len(srv.JS.euiPrefixes))
+
+	for i := range prefixes {
+		prefixes[i].JoinEUI = &srv.JS.euiPrefixes[i].EUI64
+		prefixes[i].Length = []byte{srv.JS.euiPrefixes[i].Length}
```

Comment:
I don't understand this; why use `bytes` for length?

Reply:
I did not find an int type for proto at that time.

Now I found it, but the lowest available is `uint32`, so `uint8` from euiPrefixes of `js` needs to be casted to `uint32`. Please see the updated code